common/metrics/tags.go

```diff
@@ -181,21 +177,30 @@ func ToClusterIDTag(value int32) Tag {
 	return &tagImpl{key: toCluster, value: strconv.FormatInt(int64(value), 10)}
 }
 
-// TaskQueueTag returns a new task queue tag.
-func TaskQueueTag(value string) Tag {
+// UnsafeTaskQueueTag returns a new task queue tag.
+// WARNING: Do not use this function directly in production code as it may create high number of unique task queue tag
+// values that can trouble the observability stack. Instead, use one of the following helper functions and pass a proper
+// breakdown boolean (typically based on the task queue dynamic configs):
+// - `workflow.PerTaskQueueFamilyScope`
+// - `tqid.PerTaskQueueFamilyScope`
+// - `tqid.PerTaskQueueScope`
+// - `tqid.PerTaskQueuePartitionScope`
+func UnsafeTaskQueueTag(value string) Tag {
 	if len(value) == 0 {
 		value = unknownValue
 	}
-	return &tagImpl{key: taskQueue, value: sanitizer.Value(value)}
+	return &tagImpl{key: taskQueue, value: value}
 }
 
 func TaskQueueTypeTag(tqType enumspb.TaskQueueType) Tag {
 	return &tagImpl{key: TaskTypeTagName, value: tqType.String()}
 }
 
-func WorkerBuildIdTag(buildId string) Tag {
+func WorkerBuildIdTag(buildId string, buildIdBreakdown bool) Tag {
```

Comment:
it might be helpful to put a comment here noting the dynamic config value that is most likely intended to populate `buildIdBreakdown`, but only if you feel like it, I'm not attached